pkg/cloud/aws/actuators/machine/actuator.go

```diff
@@ -133,7 +133,11 @@ func (a *Actuator) Create(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 		return errors.Wrap(err, "failed to reconcile LB attachment")
 	}
 
-	if err := a.storeMachineStatus(machine, status, true); err != nil {
+	if err := a.updateMachine(machine); err != nil {
+		glog.Errorf("failed to update machine %q in namespace %q trying to set annotation: %v", machine.Name, machine.Namespace, err)
```

Comment:
If this update fails, then `clusterctl` will still error out, instead should we add a retry that fetches the latest version of the machine object and applies the status/annotation updates?

This is going to get more important over time as we move some of the items that are only in status currently into the providerconfig/providerspec in order to allow for user-driven config.

Reply:
I don't quite follow what you're saying. Maybe this is worth opening an issue and we can fix that after this merges?

Reply:
That may do the trick, since exists will return `false` if the status has not been updated.

Reply:
Consider the `clusterctl` worklfow:
- clusterctl applies cluster.yaml
  - Create cluster
    - Update fails (results in missing annotation that clusterctl looks for)
    - Status update succeeds
  - Reconcile triggers Update, since Exists returns true.
    - Update succeeds (does not apply clusterctl required annotation)
- `clusterctl` times out waiting for machine annotation to be present

If the update failure instead throws a retry error, then it should be:
- clusterctl applies cluster.yaml
  - Create cluster
    - Update fails (results in missing annotation that clusterctl looks for)
    - Status update succeeds
  - Reconcile triggers Create, since Exists returns true.
    - Update succeeds (applies clusterctl required annotation)
    - Status update succeeds (or fails)
- `clusterctl` finds machine annotation and continues

Reply:
ok, that makes a lot of sense. i appreciate you taking the time to write that out. I'll fix it!